File: problem10/spec/p10_spec.rb
require 'spec_helper'

describe '#primes' do
  it "should return a list of all prime numbers between 2 and n" do
    n = 10
    expect(primes n).to eq ([2,3,5,7])
  end
end

describe "#is_prime?" do
  it "should return false if n <= 1" do
    n = 1
    expect(is_prime? n).to be(false)
  end

  it "should return false if n is not prime number" do
    n = 6
    expect(is_prime? n).to be(false)
  end

  it "should return true if n is a prime number" do
    n = 7
    expect(is_prime? n ).to be(true)
  end
end

describe "#list_sum" do
  it "should return the sum of the list" do
    list = [2,3,5,7]
    sum = 17
    expect(list_sum list).to eq(sum)
  end
end

File: problem10/spec/p10_spec.js
var assert = require('assert');
var app = require('../p10.js');

describe('#isPrime', function() {
  it('should return true if the number is prime', function() {
    var num = 11;
    var num2 = 7;
    assert.equal(true, app.isPrime(num));
    assert.equal(true, app.isPrime(num2));
  });

  it('should return false if the number is not prime', function() {
    var num = 12;
    var num2 = 8;
    assert.equal(false, app.isPrime(num));
    assert.equal(false, app.isPrime(num2));
  });
});

describe('#primes', function() {
  it('should return a list of primes from 2 to nth number', function() {
    var num = 10;
    assert.deepEqual([2,3,5,7], app.primes(num));
  });
});

describe('#sumList', function() {
  it('should return the sum of a list', function() {
    var list = [1,2,3];
    assert.equal(6, app.sumList(list));
  });
});
